engine/evd.ml

```diff
@@ -950,18 +950,10 @@ let mkLEvar sigma (evk, args) =
 
 let evar_handler sigma =
   let evar_expand ev = existential_expand_value0 sigma ev in
-  let qvar_irrelevant q = match UState.nf_qvar sigma.universes q with
-  | QConstant QSProp -> true
-  | QConstant QProp | QConstant QType | QVar _ -> false
-  in
+  let qvar_irrelevant q = UState.is_irrelevant sigma.universes (Sorts.RelevanceVar q) in
   let evar_irrelevant (evk, _) = match find sigma evk with
-  | EvarInfo evi ->
-    begin match evi.evar_relevance with
-    | Sorts.Relevant -> false
-    | Sorts.Irrelevant -> true
-    | Sorts.RelevanceVar q -> qvar_irrelevant q
-    end
-  | exception Not_found -> true
+  | EvarInfo evi -> UState.is_irrelevant sigma.universes evi.evar_relevance
+  | exception Not_found -> true (* XXX: Why this default ?? *)
```

Comment:
It's wrong indeed. The function used to be the flipped `evar_relevant` and this wasn't updated in b2d2e6f8787a593fc413a61252774692aaa03052.
(also probably should be anomaly)

Reply:
AFAIR you cannot put an anomaly there because many people play fast and loose with evarmaps (hm, hm).

Reply:
I guess that answers my question regarding a possble check in (to be moved) `UState.is_irrelevant`